File: v4_law_of_demeter/java/SurfShop.java
public class SurfShop {

    public static void main(String[] args) {
        SurfShop surfShop = new SurfShop();
        Customer customer = new Customer();
        surfShop.chargeCustomer(customer, 10);
        System.out.println("Done");
    }

    public void chargeCustomer(Customer c, float fee) {
        c.getCard().balance -= fee;
    }
}

class Customer {

    private Card card = new Card();

    public Card getCard() {
        return this.card;
    }

}

class Card {

    public int balance = 10;

}

File: v4_law_of_demeter/python/surfshop.py
class Card:

    balance = 10

class Customer:

    card = Card()

    def get_card(self):
        return self.card

class SurfShop:

    def charge_customer(self, c: Customer, fee: float):
        c.get_card().balance -= fee

customer = Customer()
SurfShop().charge_customer(customer, 1.0)
